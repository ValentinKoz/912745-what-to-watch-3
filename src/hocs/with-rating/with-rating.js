import React, {PureComponent} from "react";

export const withRating = (Componet) => {
  class WithRating extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        rating: 0,
        textComment: ``,
      };
      this.handleChange = this.handleChange.bind(this);
      this.handleChangeText = this.handleChangeText.bind(this);
    }

    handleChangeText(evt) {
      this.setState({textComment: evt.target.value});
    }

    handleChange(evt) {
      this.setState({rating: +evt.target.value});
    }
    render() {
      const {rating, textComment} = this.state;

      return (
        <Componet
          {...this.props}
          textComment={textComment}
          rating={rating}
          onChangeText={this.handleChangeText}
          onChangeRating={this.handleChange}
        />
      );
    }
  }
  return WithRating;
};
