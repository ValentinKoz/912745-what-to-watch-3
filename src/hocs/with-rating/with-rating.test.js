import React from "react";
import renderer from "react-test-renderer";
import {withRating} from "./with-rating.js";

const movie = {
  genre: `genre`,
  title: `title`,
  releaseDate: 1000,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  bgPoster: `img/moonrise-kingdom.jpg`,
  ratingScore: `0,0`,
  ratingLevel: `level`,
  ratingCount: 1,
  text: `...text`,
  director: `director`,
  runTime: `0h 00m`,
  starring: `starring`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  reviews: [{
    text: `...text`,
    author: `author`,
    time: `December 24, 2016`,
    rating: `0,0`
  }]
};

const MockComponent = () => <div></div>;
const MockComponentWrapped = withRating(MockComponent);

it(`MockComponentWrapped is renderer correctly`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped
          postComment={()=>{}}
          film={movie}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
