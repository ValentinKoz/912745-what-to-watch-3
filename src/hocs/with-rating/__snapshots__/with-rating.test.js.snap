// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockComponentWrapped is renderer correctly 1`] = `<div />`;
